Mathlib/Data/Complex/Basic.lean

```diff
@@ -330,22 +330,23 @@ instance addCommGroup : AddCommGroup ℂ :=
     neg_add_cancel := by intros; ext <;> simp }
 
 
+instance instNatCast : NatCast ℂ where
+  natCast := fun n => ⟨n, 0⟩
+
+@[simp]
+lemma re_ofNat (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℂ).re = ofNat(n) := rfl
+@[simp] lemma im_ofNat (n : ℕ) [n.AtLeastTwo] : (ofNat(n) : ℂ).im = 0 := rfl
+@[simp, norm_cast] lemma natCast_re (n : ℕ) : (n : ℂ).re = n := rfl
+@[simp, norm_cast] lemma natCast_im (n : ℕ) : (n : ℂ).im = 0 := rfl
+
+
 instance addGroupWithOne : AddGroupWithOne ℂ :=
   { Complex.addCommGroup with
-    natCast := fun n => ⟨n, 0⟩
-    natCast_zero := by
-      ext <;> simp [Nat.cast]
-    natCast_succ := fun _ => by ext <;> simp [Nat.cast, AddMonoidWithOne.natCast_succ]
-    intCast := fun n => ⟨n, 0⟩
-    intCast_ofNat := fun _ => by ext <;> rfl
-    intCast_negSucc := fun n => by
-      ext
-      · simp [AddGroupWithOne.intCast_negSucc]
-        show -(1 : ℝ) + (-n) = -(↑(n + 1))
-        simp [Nat.cast_add, add_comm]
-      · simp [AddGroupWithOne.intCast_negSucc]
-        show im ⟨n, 0⟩ = 0
-        rfl
+    natCast_zero := by ext <;> simp
+    natCast_succ _ := by ext <;> simp
+    intCast n := ⟨n, 0⟩
```

Comment:
I'd be inclined to split this out too for good measure

Reply:
Done